### PR TITLE
Added dynamic model

### PR DESCRIPTION
**Pull request by gprajwalpoojari**
Added Constant acceleration dynamic model

### CODE REVIEW

File: src/modules/localization/domain/dynamics/constant_acceleration_model.hpp [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+     * 
+     */
+    class ConstantAccelerationModel{
+        public:
+        /**
+         * @brief Construct a new Constant Acceleration Model object
+         * 
+         * @param[in] state_covariance_x 
+         * @param[in] state_covariance_y 
+         * @param[in] state_covariance_theta
```
Comment (aniketmpatil) [inline src/modules/localization/domain/dynamics/constant_acceleration_model.hpp]: Parameters not being taken as input can be removed

File: src/modules/localization/domain/dynamics/constant_acceleration_model.cpp [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        double dt_4 = dt_3*dt;
+
+        Q<< dt_4*noise_ax/4, 0,               0,                   dt_3*noise_ax/2, 0,               0,                   dt_2*noise_ax/2, 0,               0,
+             0,               dt_4*noise_ay/4, 0,                   0,               dt_3*noise_ay/2, 0,                   0,               dt_2*noise_ay/2, 0,
+             0,               0,               dt_4*noise_atheta/4, 0,               0,               dt_3*noise_atheta/2, 0,               0,               dt_2*noise_atheta/2,
+             dt_3*noise_ax/2, 0,               0,                   dt_2 * noise_ax, 0,               0,                   dt*noise_ax,     0,               0,
+             0,               dt_3*noise_ay/2, 0,                   0,               dt_2*noise_ay,   0,                   0,               dt*noise_ay,     0,
+             0,               0,               dt_3*noise_atheta/2, 0,               0,               dt_2*noise_atheta,   0,               0,               dt*noise_atheta,
+             dt_2*noise_ax/2, 0,               0,                   dt*noise_ax,     0,               0,                   noise_ax,        0,               0,
```
Comment (aniketmpatil) [inline src/modules/localization/domain/dynamics/constant_acceleration_model.cpp:44]: Should there be a division by 2 in element (6,1) which is dt_2*noise_ax/2 ?

**Review by aniketmpatil — CHANGES_REQUESTED** [review]
Added some comments

File: src/modules/localization/domain/dynamics/constant_acceleration_model.hpp (hunk 1 above)
Comment (gprajwalpoojari) [reply src/modules/localization/domain/dynamics/constant_acceleration_model.hpp]: good catch

File: src/modules/localization/domain/dynamics/constant_acceleration_model.cpp (hunk 2 above)
Comment (gprajwalpoojari) [reply src/modules/localization/domain/dynamics/constant_acceleration_model.cpp:44]: That is correct. But the other two are wrong. Good catch

**Review by aniketmpatil — APPROVED** [review]
Good work!

**Pull request merged**